client/src/main/java/se/fortnox/reactivewizard/client/RestClientFactory.java

```diff
@@ -4,41 +4,73 @@
 import com.google.inject.Inject;
 import com.google.inject.Provider;
 import com.google.inject.Scopes;
-import com.google.inject.TypeLiteral;
 import com.google.inject.multibindings.Multibinder;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import se.fortnox.reactivewizard.binding.AutoBindModule;
+import se.fortnox.reactivewizard.binding.scanners.HttpConfigClassScanner;
 import se.fortnox.reactivewizard.binding.scanners.JaxRsClassScanner;
 
 import java.lang.reflect.Proxy;
+import java.util.HashMap;
+import java.util.Map;
 
 public class RestClientFactory implements AutoBindModule {
 
-    private static final Logger            log = LoggerFactory.getLogger(RestClientFactory.class);
-    private final        JaxRsClassScanner jaxRsClassScanner;
+    private static final Logger                 LOG = LoggerFactory.getLogger(RestClientFactory.class);
+    private final        JaxRsClassScanner      jaxRsClassScanner;
+    private final        HttpConfigClassScanner httpConfigClassScanner;
 
     @Inject
-    public RestClientFactory(JaxRsClassScanner jaxRsClassScanner) {
+    public RestClientFactory(JaxRsClassScanner jaxRsClassScanner, HttpConfigClassScanner httpConfigClassScanner) {
         this.jaxRsClassScanner = jaxRsClassScanner;
+        this.httpConfigClassScanner = httpConfigClassScanner;
     }
 
-    private <T> Provider<T> provider(Class<T> iface, Provider<HttpClient> httpClientProvider) {
+    private <T> Provider<T> provider(Class<T> iface,
+        Provider<HttpClientProvider> httpClientProvider, Provider<? extends HttpClientConfig> httpClientConfigProvider) {
         return () -> {
-            T httpProxy = httpClientProvider.get().create(iface);
-            log.debug("Created {} for {}", Proxy.getInvocationHandler(httpProxy), iface.getName());
+
+            HttpClientConfig httpClientConfig = httpClientConfigProvider.get();
+
+            //Create client based on config and create proxy
+            T httpProxy = httpClientProvider.get().createClient(httpClientConfig).create(iface);
```

Comment:
wouldn't this result in creating a new client (with a new pool and so on) for each resource?

Reply:
...perhaps HttpClientProvider can keep a cache with client per config?

Reply:
Correct, I thought the old implementation did that as well since HttpClient was not annotated with Singleton but there the RxClientProvider is injected and that class is a Singleton.

So to mimic the old behaviour I will in HttpClientProvider create a cache with RxClientProviders using the config as a key